### ISSUE TITLE
Losses come out to be None, as if the model isn't training on the data

### ISSUE THREAD

**Issue by iHasna** [alankbi/detecto#82]
**Environment:**
Google Colab with GPU
 
This is more of an issue than a bug. I'm running everything with no errors thankfully. But its as if the model is not running or finding anything in my datasets. 
I added my images and XML files to the same folder 'Images' like it was mentioned to do in the medium article. I do want to note that some images don't have XML files cause there was no object in the image but I believe that to be fine because the 300+ images do have the object 'table'. 
I even ran the utils.xml_to_csv to make sure the xml files were being read and it was. This is the result:
![image](https://user-images.githubusercontent.com/17013945/114073476-ee7b2280-9868-11eb-94e4-d7b83e0a2cbf.png)

I then run the next few lines as shown in the tutorial:
`dataset = core.Dataset('/content/train_labels.csv', '/content/Images/')

# Customize training options

loader = core.DataLoader(dataset, batch_size=3, shuffle=True)

model = core.Model(sni)
losses = model.fit(loader, epochs=7,
                   learning_rate=0.001, verbose=True)`

And everything runs:
![image](https://user-images.githubusercontent.com/17013945/114073666-26826580-9869-11eb-86b4-a0e3ecab2fcc.png)

I then get this error:
![image](https://user-images.githubusercontent.com/17013945/114073711-3437eb00-9869-11eb-85c5-e7bfffa13f72.png)

The losses are none? As if its not training on anything? I can't seem to see where the problem is.

**Comment by alankbi** [comment]
You have to pass in a validation dataset/loader as the second argument to `model.fit` (or the keyword argument `val_dataset`) in order for it to return a list of losses. So something like this:


```python
dataset = core.Dataset('/content/train_labels.csv', '/content/Images/')
loader = core.DataLoader(dataset, batch_size=3, shuffle=True)

validation = ...  # initialize your validation dataset (either core.Dataset or core.DataLoader object)

model = core.Model(sni)
losses = model.fit(loader, validation, epochs=7, learning_rate=0.001, verbose=True)
```

**Comment by iHasna** [comment]
Ah okay thanks for letting me know. However, when I try to predict on an image I used for training I get empty lists. Code below:
![image](https://user-images.githubusercontent.com/17013945/114205027-7f143a00-991f-11eb-9b44-07e76b49f611.png)

(idk why it turns it yellow, its originally white.)

![image](https://user-images.githubusercontent.com/17013945/114205111-94896400-991f-11eb-8beb-734c60e6ed3c.png)

![image](https://user-images.githubusercontent.com/17013945/114205158-a1a65300-991f-11eb-94c6-02356bbed028.png)

**Comment by alankbi** [comment]
That likely means the model wasn't trained well enough in order to detect any objects. You could try printing or plotting the returned list of losses (if you have a validation dataset specified) to see if the loss is continuously decreasing (maybe you need more epochs), or potentially add additional images or augmentations to strengthen your dataset.
